packages/core/src/utils.js

```diff
@@ -244,7 +245,12 @@ function computeDefaults(
           (formData || {})[key],
           includeUndefinedValues
         );
-        if (includeUndefinedValues || computedDefault !== undefined) {
+        if (
+          includeUndefinedValues ||
+          (typeof computedDefault === "object"
```

Comment:
This is hard to parse. Can we split it into two if statements?

if (typeof computedDefault === "object") {// if statement here}
...
else {// other if statement here }

And also comment out what each branch means